LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -219,5 +245,31 @@ public void RegisterStudentsForClasses()
             throw new NotImplementedException();
         }
 
+        #region Reports
+        void ReportHeader(string actionName)
+        {
+            Console.WriteLine($"******{actionName} is started******\n");
+        }
+
+        void ReportFooter(string actionName)
+        {
+            Console.WriteLine($"******{actionName} is finished******\n");
+        }
+
+        void ReportItem(string itemName, string actionName, int itemIndex)
+        {
+            Console.WriteLine($" {actionName} {itemIndex}\n{itemName}");
+        }
+
+        void ReportSummary(string summary, int count)
+        {
+            Console.WriteLine($"''''''Generated {count} {summary}''''''\n");
+        }
+
+        void ReportError(string actionName, string actionItem)
```

Comment:
The names of the parameters are misleading and the method is not generic. You should be able to handle all possible failures here. It should be something like:
        void ReportError(string missingResource, string failedAction)
        {
            Console.WriteLine("--------- You do not have the appropriate {0} to  {1} ---------\n", missingResource, failedAction);
        }